packages/assets-controllers/src/assetsUtil.ts

```diff
@@ -245,6 +244,7 @@ export function getIpfsCIDv1AndPath(ipfsUrl: string): {
   const cid = index !== -1 ? url.substring(0, index) : url;
   const path = index !== -1 ? url.substring(index) : undefined;
 
+  const { CID } = await import('multiformats');
```

Comment:
`multiformats` is an ESM-only package, and we are replacing its import statement with dynamic import syntax.

This causes the following changes:

- `getIpfsCIDv1AndPath`, `getFormattedIpfsUrl` are now async functions (this is a breaking change for `assets-controllers`).
- The `--experimental-vm-modules` flag needs to be prepended to our build scripts that run `jest`, with [accompanying updates to our yarn constraints](https://github.com/MetaMask/core/pull/3645#discussion_r1667240716).

Reply:
How did this work previously if it's ESM-only?

Reply:
So far as I can tell, our previous `moduleResolution` option `node`/`node10` will happily pass ESM-only packages into `require` calls without raising any warnings. 

So it's likely the warnings we're now seeing with `node16` should have been there all along. The package migrated to ESM 4 years ago (https://github.com/multiformats/js-multiformats/commit/f9251959245a2790e3d3cff216d3d3a8ca896d42#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519).

At runtime, either the module was interoperable by coincidence (not quite sure if this is possible or likely -- maybe if there's no ESM-specific syntax?), or it was breaking and we didn't notice.